Testing feature file

raeeschachar opened the pull request:
@benpatterson @jzoldak 
Please review this. Is this what you guys are expecting as a first audit? 
This is the courseware feature file.
Example: registration.feature and annotatable.feature files can be removed as they are already covered in bok choy tests? For the rest we need to add tests?
@Muddasser FYI

raeeschachar commented:
jenkins run bokchoy

benpatterson commented:
@raeeschachar Yes that is great. I think that we don't need to put those comments in the code though...we should take each of your findings and make them a to-do. For example, the two scenarios that are covered in other tests, you could put in a PR to remove the .feature scenarios for them.

The other parts, I don't know the answer. I'd have to do some work to figure out who the right person would be to ask, but perhaps you could do some of that work instead of me. Once you get a clear idea if it's covered or not, you could take the right next step.

Does that help?

raeeschachar commented:
@benpatterson @jzoldak @Muddasser 
Fixed this PR, please review.
This only covers one part i.e courseware/features
- Removed the scenario/feature file which are already covered in bok choy tests
- Noted a to do on tests that we need to write

jzoldak commented:
@raeeschachar 
If annotable.py and events.py in the same directory as the annotable.feature and events.feature directory have methods that were only used by those feature files, let's remove them also in this PR.

benpatterson commented:
Ah good point, JZ.

Once that's done and builds pass, :+1: from me

raeeschachar commented:
jenkins run lettuce

raeeschachar commented:
jenkins run lettuce

raeeschachar commented:
@jzoldak 
Comments addressed but it is failing Lettuce tests that it shouldn't, is there anything else to take care off for this?

jzoldak commented:
@raeeschachar from https://build.testeng.edx.org/job/edx-platform-test-subset/364483/console it looks like some other lettuce test(s) are using these 2 steps which were in events.py. Probably you need to either move them to a common file, or leave events.py in place.
```
00:21:58.214 You can implement step definitions for undefined steps with these snippets:
00:21:58.214 
00:21:58.214 # -*- coding: utf-8 -*-
00:21:58.214 from lettuce import step
00:21:58.215 
00:21:58.215 @step(u'And a "([^"]*)" browser event is emitted')
00:21:58.215 def and_a_group1_browser_event_is_emitted(step, group1):
00:21:58.215     assert False, 'This step must be implemented'
00:21:58.215 @step(u'And a "([^"]*)" server event is emitted')
00:21:58.216 def and_a_group1_server_event_is_emitted(step, group1):
00:21:58.216     assert False, 'This step must be implemented'
```

raeeschachar commented:
@jzoldak 
I restored the file so that once we cover the tests in bokchoy we can go ahead and remove these files. Let me know if you think we should do it the other way, otherwise a thumbs up please :)
Thanks for all your help.

jzoldak commented:
sounds good to me @raeeschachar :+1:

benpatterson commented:
:+1:
